### ISSUE TITLE
Why  bcbio_nextgen.s3.amazonaws.com  is not working

### ISSUE THREAD

**Issue by cenielyj** [chapmanb/bcbio-nextgen#1211]
Dear Chapmanb,
I have tried 10 times, that I always get this error message, 

--2016-02-03 19:10:52--  http://bcbio_nextgen.s3.amazonaws.com/GA4GH_problem_regions.zip
Resolving bcbio_nextgen.s3.amazonaws.com (bcbio_nextgen.s3.amazonaws.com)... 220.165.8.172, 220.165.8.174
Connecting to bcbio_nextgen.s3.amazonaws.com (bcbio_nextgen.s3.amazonaws.com)|220.165.8.172|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2016-02-03 19:11:24 ERROR 404: Not Found.

may I ask whether the server is down?

**Comment by cenielyj** [comment]
also I have been stucked there for a while, is there anyway to bypass download this and move on to the next file?

**Comment by guillermo-carrasco** [comment]
Hi @cenielyj ,

The file is there and the server seems to be up, its actually an amazon S3 bucket. Just tried:

```
$> wget http://bcbio_nextgen.s3.amazonaws.com/GA4GH_problem_regions.zip

--2016-02-03 12:39:25--  http://bcbio_nextgen.s3.amazonaws.com/GA4GH_problem_regions.zip
Resolving bcbio_nextgen.s3.amazonaws.com... 54.231.11.144
Connecting to bcbio_nextgen.s3.amazonaws.com|54.231.11.144|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 93336848 (89M) [application/zip]
Saving to: 'GA4GH_problem_regions.zip'

GA4GH_problem_regions.zip                    100%[==============================================================================================>]  89.01M  3.10MB/s   in 29s

2016-02-03 12:39:55 (3.10 MB/s) - 'GA4GH_problem_regions.zip' saved [93336848/93336848]
```

Can you confirm that you're not behind any firewall that could be blocking the connection? Were you able to download other files from Amazon? Do you get a response from `ping http://bcbio_nextgen.s3.amazonaws.com`?

Thanks!

**Comment by chapmanb** [comment]
Sorry about the download issues. I'm unfortunately not familiar with download restrictions in China but imagine these are causing the underlying problems. This dataset is hosted on Amazon S3 so is up and available as Guillermo mentioned.

Right now the bcbio installation pulls resources from a large number of locations. Some of these we manage ourselves, like the S3 buckets on Amazon, and others are from publicly available resources. As implemented currently it's not going to be straightforward to manually download and retrieve all these resources outside of the install script, and I'd hate to suggest doing that amount of work.

One of our goals is to make the data installation process be modular so you can get specific subsets of the data on demand, and this would make it easier to install on-demand sections of data. Apologies this isn't done now but hopefully that'll improve the process in the future. In the short term if you're able to work around the firewall/restriction/bandwidth issues that would be your best approach to getting the full install finished. Hope this helps some.

**Comment by cenielyj** [comment]
Thanks I will try again

**Comment by cenielyj** [comment]
Hi @guillermo-carrasco 
I can get a ping and I also can download directly from a browser, however just cannot get pass in terminal.

**Comment by adamcornish** [comment]
I'm having a similar problem when I try to install with GRCh37 selected as the genome of choice. It seems that, while the bwa and bowtie indexes are available, the rtg index is missing and that's causing the installation to abort. To verify this, I tried a simple wget command on the file and get a 404 error:

`wget https://s3.amazonaws.com/biodata/genomes/GRCh37-rtg.tar.xz
--2016-02-17 15:58:46--  https://s3.amazonaws.com/biodata/genomes/GRCh37-rtg.tar.xz
Resolving s3.amazonaws.com... 54.231.49.172
Connecting to s3.amazonaws.com|54.231.49.172|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2016-02-17 15:58:46 ERROR 404: Not Found.`

The error bcbio generates is: `Warning: local() encountered an error (return code 8) while executing 'wget --continue --no-check-certificate -O GRCh37-rtg.tar.xz 'https://s3.amazonaws.com/biodata/genomes/GRCh37-rtg.tar.xz''`

For what it's worth, I'm in the US and have tried this from multiple locations.

**Comment by chapmanb** [comment]
Adam;
Sorry about the issues. The 404 for this file is expected, as we provide some pre-built downloads for genome indices but not for the GRCh37 rtg indices since they build quickly locally. It should recover from this and continue processing to build the index locally. Is it possible you're getting some error after this? That would be the casual problem we can debug. You could post the whole log after the message above here or as a Gist (https://gist.github.com/) which might help us identify the underlying issue. Hope this helps.

**Comment by adamcornish** [comment]
Hi @chapmanb,

Thanks for the quick response. Looks like the error that caused the installation to abort was indeed due to something amiss on my end: the RTG.jar file had somehow been removed from my ~/.conda folder during installation and that was causing some linking issues to the bcbio_install folder I created (I think this was the result of previous failed attempts at installing the pipeline due to the wonkiness of the cluster I'm working on). However, once I removed the ~/.conda folder and started from scratch, the installation script happily trundled along past the previous sticking point.

Cheers,

Adam
